src/main/java/arb/logic/commands/project/MarkProjectCommand.java

```diff
@@ -0,0 +1,65 @@
+package arb.logic.commands.project;
+
+import static arb.model.Model.PREDICATE_SHOW_ALL_PROJECTS;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import arb.commons.core.Messages;
+import arb.commons.core.index.Index;
+import arb.logic.commands.Command;
+import arb.logic.commands.CommandResult;
+import arb.logic.commands.exceptions.CommandException;
+import arb.model.ListType;
+import arb.model.Model;
+import arb.model.project.Project;
+
+/**
+ * Marks a project identified using it's displayed index from the address book.
+ */
+public class MarkProjectCommand extends Command {
+
+    public static final String COMMAND_WORD = "mark";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Marks the project identified by the index number used in the displayed project list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_MARK_PROJECT_SUCCESS = "Marked Project: %1$s";
+
+    private final Index targetIndex;
+
+    public MarkProjectCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model, ListType currentListBeingShown) throws CommandException {
+        requireNonNull(model);
+        List<Project> lastShownList = model.getFilteredProjectList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PROJECT_DISPLAYED_INDEX);
+        }
+
+        if (currentListBeingShown != ListType.PROJECT) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PROJECT_DISPLAYED_INDEX);
+        }
+
+        Project projectToMark = lastShownList.get(targetIndex.getZeroBased());
+        model.markProject(projectToMark);
+
+        model.setProject(projectToMark, projectToMark);
+        //projectToMark.markAsDone();
```

Comment:
Sorry I forgot, I meant to leave this statement in comment 
Because earlier you recommended putting this statement inside but turns out the statement projectToMark.markAsDone() in UniqueProjectList.java is the correct one? Which I found it confusing. I tried to debug it and find out the reason why but to no avail haha